src/Daemon/BaseDaemon.cpp

```diff
@@ -1013,7 +1013,15 @@ void BaseDaemon::setupWatchdog()
         /// If streaming compression of logs is used then we write watchdog logs to cerr
         if (config().getRawString("logger.stream_compress", "false") == "true")
         {
-            Poco::AutoPtr<OwnPatternFormatter> pf = new OwnPatternFormatter;
+            Poco::AutoPtr<OwnPatternFormatter> pf;
+            if (config().has("logger.json"))
+            {
+                pf = new OwnJSONPatternFormatter;
+            }
+            else
+            {
+                pf = new OwnPatternFormatter(true);
+            }
```

Comment:
This way it looks better
```suggestion
                pf = new OwnJSONPatternFormatter;
            else
                pf = new OwnPatternFormatter(true);

```

Reply:
Changes applied.